de-lib/src/main/java/org/iplantc/de/commons/client/views/dialogs/ClipboardCopyEnabledDialog.java

```diff
@@ -0,0 +1,73 @@
+package org.iplantc.de.commons.client.views.dialogs;
+
+import org.iplantc.de.client.util.CopyToClipboardUtil;
+import org.iplantc.de.commons.client.info.ErrorAnnouncementConfig;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.user.client.ui.HorizontalPanel;
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.TextBox;
+
+import com.sencha.gxt.widget.core.client.button.TextButton;
+import com.sencha.gxt.widget.core.client.container.VerticalLayoutContainer;
+
+/**
+ * A dialog that provides ability to copy text to clipboard from a textbox
+ *
+ * Created by sriram on 11/3/17.
+ */
+public class ClipboardCopyEnabledDialog extends IPlantDialog {
+
+
+    public interface Appearance {
+        String copyPasteInstructions();
+
+        String copyToClipboard();
+
+        String textboxWidth();
+
+        String textboxHeight();
+
+        String errorCopying();
+    }
+
+    protected TextBox textBox;
+    protected TextButton copyBtn;
+
+    Appearance appearance = GWT.create(Appearance.class);
+
+    public ClipboardCopyEnabledDialog() {
```

Comment:
Can this class be injected?

Reply:
Actually this class is meant to be extended like `IPlantDialog`. So the class thats extending this one can do the injection.